### PR TITLE
dbt: init at 1.5.3

### LINKED ISSUE #137292: dbt CLI (data build tool)
**Project description**
Open-source CLI core of widely-used data transformation tool, written in Python.

**Metadata**

* homepage URL: [getdbt.com/docs]()
* source URL: [github.com/dbt-labs/dbt]()
* license: Apache 2.0
* platforms: unix

### PR DESCRIPTION
**Pull request by tjni**
###### Description of changes

DBT is an open-source CLI core of widely-used data transformation tool.

These derivations are co-authored by @mausch and picks up where https://github.com/NixOS/nixpkgs/pull/220305 left off. @mausch, I kept you as a maintainer of all the packages you packaged in the PR. Feel free to add yourself to any others too!

For reviewers: the part that I could use the most feedback on is the last commit, which adds the `dbt` application along with a `withAdapters` function to add adapters to various data sources (which are just Python packages). How it should be used is in the `meta.longDescription` of the `dbt-core` package.

Supercedes https://github.com/NixOS/nixpkgs/pull/146767 and resolves https://github.com/NixOS/nixpkgs/issues/137292.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/minimal-snowplow-tracker/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    requests
+    six
+  ];
+
+  pythonImportsCheck = [
+    "snowplow_tracker"
+  ];
+
+  meta = with lib; {
+    description = "A minimal snowplow event tracker for Python";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/minimal-snowplow-tracker/default.nix:28]:
```suggestion
    description = "Minimal snowplow event tracker";
```

File: pkgs/development/python-modules/dbt-extractor/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    rustPlatform.cargoSetupHook
+    rustPlatform.maturinBuildHook
+  ];
+
+  buildInputs = lib.optionals stdenv.isDarwin [
+    libiconv
+  ];
+
+  # no python tests exist
+  dontUseSetuptoolsCheck = true;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dbt-extractor/default.nix:37]: Wouldn't we want to use doCheck = false ?

File: pkgs/development/python-modules/minimal-snowplow-tracker/default.nix (hunk 1 above)
Comment (tjni) [reply pkgs/development/python-modules/minimal-snowplow-tracker/default.nix]: done

File: pkgs/development/python-modules/dbt-extractor/default.nix (hunk 2 above)
Comment (tjni) [reply pkgs/development/python-modules/dbt-extractor/default.nix]: Good call.

File: pkgs/development/python-modules/minimal-snowplow-tracker/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    six
+  ];
+
+  pythonImportsCheck = [
+    "snowplow_tracker"
+  ];
+
+  meta = with lib; {
+    description = "Minimal snowplow event tracker";
+    homepage = "https://pypi.org/project/minimal-snowplow-tracker";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/minimal-snowplow-tracker/default.nix:29]: Do we have the source code somewhere?

File: pkgs/development/python-modules/minimal-snowplow-tracker/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-rKv3Vy2w5/XL9pg9SV7vVAgfcb45IzDrOq25zLOdqqQ=";
+  };
+
+  propagatedBuildInputs = [
+    requests
+    six
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/minimal-snowplow-tracker/default.nix:22]: Do we have tests to run?

File: pkgs/development/python-modules/minimal-snowplow-tracker/default.nix (hunk 3 above)
Comment (tjni) [reply pkgs/development/python-modules/minimal-snowplow-tracker/default.nix]: I found a repository for it and updated the link.

File: pkgs/development/python-modules/minimal-snowplow-tracker/default.nix (hunk 4 above)
Comment (tjni) [reply pkgs/development/python-modules/minimal-snowplow-tracker/default.nix:22]: The source code has some tests that might have been left over from when it was forked, but the repo has no tags of its own to mark its releases, and the tests are not included in the PyPI release. I disabled `doCheck` and left a comment saying that there are no tests.

**Pull request merged**